lib/src/Pos/Worker/Block.hs

```diff
@@ -211,7 +211,7 @@ dropObftEbb genesisConfig txpConfig = do
     tipHeader <- DB.getTipHeader
     case tipHeader of
         BlockHeaderMain _      -> pure ()
-        BlockHeaderGenesis _   -> do
+        BlockHeaderGenesis _   -> unless (tipHeader ^. epochIndexL == 0) $ do
```

Comment:
We could potentially doc why this `unless` is necessary. I guess a comment similar to your commit message might be sufficient. Maybe something like:

```
-- If we're starting up a chain in OBFT mode, we need to ensure that we don't 
-- rollback the actual genesis block out of existence (i.e. the EBB at epoch 0)
```

What do you think?

Reply:
Good point, done @intricate !